python_modules/dagster/dagster/_utils/__init__.py

```diff
@@ -211,95 +214,71 @@ def mkdir_p(path: str) -> str:
             raise
 
 
-# TODO: Make frozendict generic for type annotations
-# https://github.com/dagster-io/dagster/issues/3641
-class frozendict(dict):
-    def __readonly__(self, *args, **kwargs):
-        raise RuntimeError("Cannot modify ReadOnlyDict")
+def hash_named_tuple(named_tuple: NamedTuple) -> int:
```

Comment:
lets put these `hash_` functions under direct test to encode what exact behaviors we expect them to have ie dicts hash the same despite order, lists do not etc